flink-connectors/flink-connector-aws-base/src/test/java/org/apache/flink/connector/aws/util/AWSGeneralUtilTest.java

```diff
@@ -0,0 +1,536 @@
+
+        assertTrue(credentialsProvider instanceof ProfileCredentialsProvider);
+
+        AwsCredentials credentials = credentialsProvider.resolveCredentials();
+        assertEquals("22222222222222222222", credentials.accessKeyId());
+        assertEquals("wJalrXUtnFEMI/K7MDENG/bPxRfiCY2222222222", credentials.secretAccessKey());
+    }
+
+    @Test
+    public void testCreateNettyAsyncHttpClientWithDefaults() throws Exception {
+        NettyNioAsyncHttpClient.Builder builder = NettyNioAsyncHttpClient.builder();
+
+        SdkAsyncHttpClient httpClient = AWSGeneralUtil.createAsyncHttpClient(builder);
+        NettyConfiguration nettyConfiguration = TestUtil.getNettyConfiguration(httpClient);
+
+        SdkAsyncHttpClient httpDefaultClient = NettyNioAsyncHttpClient.create();
+        NettyConfiguration nettyDefaultConfiguration =
+                TestUtil.getNettyConfiguration(httpDefaultClient);
+
+        assertEquals(
+                AWSGeneralUtil.CONNECTION_ACQUISITION_TIMEOUT.toMillis(),
+                nettyConfiguration.connectionAcquireTimeoutMillis());
+        assertEquals(
+                nettyDefaultConfiguration.connectionTtlMillis(),
+                nettyConfiguration.connectionTtlMillis());
+        assertEquals(
+                nettyDefaultConfiguration.connectTimeoutMillis(),
+                nettyConfiguration.connectTimeoutMillis());
+        assertEquals(
+                nettyDefaultConfiguration.idleTimeoutMillis(),
+                nettyConfiguration.idleTimeoutMillis());
+        assertEquals(
+                AWSGeneralUtil.HTTP_CLIENT_MAX_CONCURRENCY, nettyConfiguration.maxConnections());
+        assertEquals(
+                nettyDefaultConfiguration.maxPendingConnectionAcquires(),
+                nettyConfiguration.maxPendingConnectionAcquires());
+        assertEquals(
+                AWSGeneralUtil.HTTP_CLIENT_READ_TIMEOUT.toMillis(),
+                nettyConfiguration.readTimeoutMillis());
+        assertEquals(
+                nettyDefaultConfiguration.reapIdleConnections(),
+                nettyConfiguration.reapIdleConnections());
+        assertEquals(nettyDefaultConfiguration.tcpKeepAlive(), nettyConfiguration.tcpKeepAlive());
+        assertEquals(
+                nettyDefaultConfiguration.tlsKeyManagersProvider(),
+                nettyConfiguration.tlsKeyManagersProvider());
+        assertEquals(
+                nettyDefaultConfiguration.tlsTrustManagersProvider(),
+                nettyConfiguration.tlsTrustManagersProvider());
+        assertEquals(
+                AWSGeneralUtil.TRUST_ALL_CERTIFICATES, nettyConfiguration.trustAllCertificates());
+        assertEquals(
+                nettyDefaultConfiguration.writeTimeoutMillis(),
+                nettyConfiguration.writeTimeoutMillis());
+        assertEquals(
+                AWSGeneralUtil.HTTP_PROTOCOL,
+                nettyConfiguration.attribute(SdkHttpConfigurationOption.PROTOCOL));
```

Comment:
nit: These would be best as single tests. Best practise is to make tests test a single thing, and ideally have a single assertion. This way we fail fast, and get all failures immediately.